segmentation/include/pcl/segmentation/euclidean_cluster_comparator.h

```diff
@@ -42,158 +42,232 @@
+  class EuclideanClusterComparator : public experimental::EuclideanClusterComparator<PointT, PointLT>
   {
     protected:
 
-      using pcl::Comparator<PointT>::input_;
+      using experimental::EuclideanClusterComparator<PointT, PointLT>::exclude_labels_;
 
     public:
-      using typename Comparator<PointT>::PointCloud;
-      using typename Comparator<PointT>::PointCloudConstPtr;
-      
-      typedef typename pcl::PointCloud<PointLT> PointCloudL;
-      typedef typename PointCloudL::Ptr PointCloudLPtr;
-      typedef typename PointCloudL::ConstPtr PointCloudLConstPtr;
 
-      typedef boost::shared_ptr<EuclideanClusterComparator<PointT, PointLT> > Ptr;
-      typedef boost::shared_ptr<const EuclideanClusterComparator<PointT, PointLT> > ConstPtr;
+      typedef typename pcl::PointCloud<PointNT> PointCloudN;
+      typedef typename PointCloudN::Ptr PointCloudNPtr;
+      typedef typename PointCloudN::ConstPtr PointCloudNConstPtr;
 
-      typedef std::map<uint32_t, bool> ExcludeLabelMap;
-      typedef boost::shared_ptr<ExcludeLabelMap> ExcludeLabelMapPtr;
-      typedef boost::shared_ptr<const ExcludeLabelMap> ExcludeLabelMapConstPtr;
+      typedef boost::shared_ptr<EuclideanClusterComparator<PointT, PointNT, PointLT> > Ptr;
+      typedef boost::shared_ptr<const EuclideanClusterComparator<PointT, PointNT, PointLT> > ConstPtr;
+
+      using experimental::EuclideanClusterComparator<PointT, PointLT>::setExcludeLabels;
 
       /** \brief Empty constructor for EuclideanClusterComparator. */
+      PCL_DEPRECATED ("EuclideanClusterComparator will no longer be templated on a PointNormal type in future minor release")
       EuclideanClusterComparator ()
-        : distance_threshold_ (0.005f)
-        , depth_dependent_ ()
-        , z_axis_ ()
-      {
-      }
-      
-      /** \brief Destructor for EuclideanClusterComparator. */
-      virtual
-      ~EuclideanClusterComparator ()
-      {
-      }
+        : normals_ ()
+        , angular_threshold_ (0.0f)
+      {}
 
-      virtual void 
-      setInputCloud (const PointCloudConstPtr& cloud)
-      {
-        input_ = cloud;
-        Eigen::Matrix3f rot = input_->sensor_orientation_.toRotationMatrix ();
-        z_axis_ = rot.col (2);
-      }
-      
-      /** \brief Set the tolerance in meters for difference in perpendicular distance (d component of plane equation) to the plane between neighboring points, to be considered part of the same plane.
-        * \param[in] distance_threshold the tolerance in meters 
-        * \param depth_dependent
-        */
+      /** \brief Provide a pointer to the input normals.
+       * \param[in] normals the input normal cloud
+       */
       inline void
-      setDistanceThreshold (float distance_threshold, bool depth_dependent)
-      {
-        distance_threshold_ = distance_threshold;
-        depth_dependent_ = depth_dependent;
-      }
+      PCL_DEPRECATED ("EuclideanClusterComparator no longer makes use of normals.")
```

Comment:
Actually, it never used normals in the first place. I think it can be useful to let the users know this. Otherwise the line of thinking when someone sees this message is: "Oh crap, what should I do now? I still need to compare normals. Is there another class now? Which one, how to find it?"

So perhaps a better deprecation message would be something along these lines: "EuclideadClusterComparator never actually used normals and angular threshold, this function has no effect on the behavior of the comparator. Therefore it is deprecated and will be removed in future releases."

Same for other functions.